### PR TITLE
feat: [CI-5359]: add error logs

### PR DESCRIPTION
**Pull request by harshjain12**
## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>
  
- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>
  
  You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`
  
- Compile: `trigger compile`
- CodeFormat: `trigger codeformat`
- MessageMetadata: `trigger messagecheck`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- runDockerizationCheck: `trigger dockerizationcheck`
- runAuthorCheck: `trigger authorcheck`
- Checkstyle: `trigger checkstyle`
- PMD: `trigger pmd`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/36775)

### CODE REVIEW

File: 960-api-services/src/main/java/io/harness/cistatus/service/GithubServiceImpl.java [hunk 1]
```diff
@@ -87,8 +87,14 @@ public boolean sendStatus(GithubAppConfig githubAppConfig, String token, String
 
     try {
       Response<StatusCreationResponse> githubStatusCreationResponseResponse =
-          getGithubClient(githubAppConfig).createStatus(getAuthToken(token), owner, repo, sha, bodyObjectMap).execute();
-
+          getGithubClient(githubAppConfig)
+              .createStatus(getAuthToken(token), owner, repo, "DFd", bodyObjectMap)
```
Comment (shubham149) [inline 960-api-services/src/main/java/io/harness/cistatus/service/GithubServiceImpl.java]: Dfd -> sha

File: 960-api-services/src/main/java/io/harness/cistatus/service/GithubServiceImpl.java [hunk 2]
```diff
@@ -106,6 +112,7 @@ public String findPR(String apiUrl, String token, String owner, String repo, Str
       if (response.isSuccessful()) {
         return new ObjectMapper().writerWithDefaultPrettyPrinter().writeValueAsString(response.body());
       } else {
+        log.error("Failed to find pr error {}, message {}", response.errorBody().string(), response.message());
```
Comment (shubham149) [inline 960-api-services/src/main/java/io/harness/cistatus/service/GithubServiceImpl.java:113]: add repo for pr in err message along with pr number

File: 960-api-services/src/main/java/io/harness/cistatus/service/GithubServiceImpl.java [hunk 3]
```diff
@@ -128,6 +135,7 @@ public JSONObject mergePR(String apiUrl, String token, String owner, String repo
         json.put("message", ((LinkedHashMap) response.body()).get("message"));
         return json;
       } else {
+        log.error("Failed to merge pr error {}, message {}", response.errorBody().string(), response.message());
```
Comment (shubham149) [inline 960-api-services/src/main/java/io/harness/cistatus/service/GithubServiceImpl.java:136]: ditto

File: 960-api-services/src/main/java/io/harness/cistatus/service/GithubServiceImpl.java [hunk 4]
```diff
@@ -150,6 +158,8 @@ public List<GitPollingWebhookData> getWebhookRecentDeliveryEvents(
           delivery -> delivery.getStatusCode() != HttpStatus.SC_ACCEPTED);
 
       if (response.isSuccessful()) {
+        log.error(
+            "Failed to get webhook details error {}, message {}", response.errorBody().string(), response.message());
```
Comment (shubham149) [inline 960-api-services/src/main/java/io/harness/cistatus/service/GithubServiceImpl.java:160]: ditto

File: 960-api-services/src/main/java/io/harness/cistatus/service/GithubServiceImpl.java [hunk 5]
```diff
@@ -193,6 +203,8 @@ private List<GitPollingWebhookData> getWebhookDeliveryFullEvents(
 
                           .build());
         } else {
+          log.error("Failed to get webhook details error {}, message {}", fullWebhookResponse.errorBody().string(),
```
Comment (shubham149) [inline 960-api-services/src/main/java/io/harness/cistatus/service/GithubServiceImpl.java:204]: ditto

**Comment by diptiman-adak** [comment]
Quality gate Quality gate: **_Passed_** 
 


            
 No issues Everything looks good
 [Details](https://app.embold.io/organization/gh/harness/repositories/7752dbb3099d2dba3b5d37f5931a5e4c/pull-request/36775)

**Review by shubham149 — APPROVED** [review]
> Reviewed 2 of 3 files at r1, 1 of 1 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/36775)* status: :shipit: complete! ✓

**Comment by harshjain12** [comment]
trigger messagecheck

**Pull request merged**